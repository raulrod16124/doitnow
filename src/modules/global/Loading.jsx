import React from "react";

import SpinnerLoading from "./../../assets/Spin-1s-200px.svg";

export const Loading = () => {
  return (
    <div className="loading">
      <div className="content-spinner">
        <img src={SpinnerLoading} alt="Loading spinner" />
      </div>
    </div>
  );
};
