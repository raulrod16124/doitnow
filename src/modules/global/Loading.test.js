import { render, screen } from "@testing-library/react";

import { Loading } from "./Loading";

describe("Testing Loading Compoent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(<Loading />);
  });

  test("should render the Loading component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should find the img Role of the spinner", () => {
    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  test("should find the text Loading", () => {
    expect(screen.getByText("Loading")).toBeInTheDocument();
  });
});
