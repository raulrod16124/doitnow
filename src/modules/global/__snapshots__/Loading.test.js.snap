// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading Compoent should render the Loading component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loading"
      >
        <div
          class="content-spinner"
        >
          <img
            alt="Loading spinner"
            src="Spin-1s-200px.svg"
          />
          <h2
            class="text"
          >
            Loading
          </h2>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loading"
    >
      <div
        class="content-spinner"
      >
        <img
          alt="Loading spinner"
          src="Spin-1s-200px.svg"
        />
        <h2
          class="text"
        >
          Loading
        </h2>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
